migration-engine/connectors/migration-connector/src/destructive_changes_checker.rs

```diff
@@ -1,25 +1,45 @@
+use crate::ConnectorResult;
 use std::marker::PhantomData;
 
 pub trait DestructiveChangesChecker<T>: Send + Sync + 'static
 where
-    T: Send + Sync + 'static,
+    T: 'static,
```

Comment:
There's a mismatch between what we need here (the Send + Sync bounds aren't necessary) and what we need in `apply_migration.rs`, i.e. if we start using the method there, we need to add Send + Sync bounds to the arguments of a couple of methods. I don't know the more general context, is there any reason we need these trait bounds here?

Reply:
If I remember it right, it had something to do with this. https://github.com/prisma/prisma-engine/blob/master/migration-engine/core/src/api/rpc.rs#L197

Reply:
The commit that added the bounds is 5e01786d78a961388c69fb048b88a8c3c52611c9

```
Julius de Bruijn, a month ago   (August 6th, 2019 7:48pm)

Performance fixes for the migration-engine

* Start a stdin listener service instead of handling one command per
process
* Keep the connection stored in the service
* Abstract the service enough to not get flabbergasted with type params
everywhere

```

Sounds legit, so I think I'll propagate the bounds in `apply_migration.rs`.

Reply:
done in 5a4a0ddaf938f710fec36900d72555429bab5dd8